File: inhouse/simple_cpg.py
# Old CpG experiment (Python)
# - Counts the number of non-overlapping [CG]+ regions 

import sys

def is_cpg(s):
   return s == 'C' or s == 'G' 

def cpg_count(s):
   i = 0
   total = 0
   while i < len(s):
      if is_cpg(s[i]):
         j = i + 1
         while j < len(s) and is_cpg(s[j]):
            j += 1
         total += 1
         i = j + 1
      else: i += 1
   return total

f = open(sys.argv[1], 'r')
cnt = 0
for l in f:
   cnt += cpg_count(l.strip())
print cnt

File: inhouse/simple_cpg.jl
# Old CpG experiment (Julia)
# - Counts the number of non-overlapping [CG]+ regions 

function is_cpg(s)
   s == 'C' || s == 'G' 
end

function cpg(s)
   i, cnt = 1, 0
   while i <= length(s)
      if is_cpg(s[i]) 
         j = i + 1
         while j <= length(s) && is_cpg(s[j])
            j += 1
         end
         cnt += 1
         i = j + 1
      else
         i += 1
      end
   end
   cnt
end

function main(arg)
   f = open(arg)
   cnt = 0
   for l in eachline(f)
      cnt += cpg(l)
   end
   println("$cnt")
end

main(ARGS[1])